src/resmom/mom_main.c

```diff
@@ -8309,15 +8309,39 @@ main(int argc, char *argv[])
 #if defined(RLIM64_INFINITY)
 	{
 		struct rlimit64 rlimit;
+		int curerror;
 
 		rlimit.rlim_cur = RLIM64_INFINITY;
 		rlimit.rlim_max = RLIM64_INFINITY;
 
 		(void)setrlimit64(RLIMIT_CPU,   &rlimit);
 		(void)setrlimit64(RLIMIT_FSIZE, &rlimit);
 		(void)setrlimit64(RLIMIT_DATA,  &rlimit);
-		(void)getrlimit64(RLIMIT_STACK, &orig_stack_size); /* get for later */
-		(void)setrlimit64(RLIMIT_STACK, &rlimit);
+		if (getrlimit64(RLIMIT_STACK, &orig_stack_size) != -1) {
+			if((orig_stack_size.rlim_cur != RLIM_INFINITY) && (orig_stack_size.rlim_cur < MIN_STACK_LIMIT)) {
```

Comment:
will it be RLIM_INFINITY or RLIM64_INFINITY for 64 bit machines?

Reply:
Thanks for your reviews @arungrover 
Good catch, I have addressed your comment.